src/js/legend.js

```diff
@@ -25,19 +25,61 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
             // dataSet accepts:
             // 1. an array of objects. Must contain "id", "value" and "label" variables.
             // Example: [{id: string, value: number, label: string} ... ]
-            dataSet: []
+            dataSet: [],
+            total: {
+                // value: number,
+                // percentage: number
+            }
         },
+        modelRelay: [{
```

Comment:
Recall... that grades are not limited to the "bounded functions" that would be recognised by an object-oriented theoretician : P
As well as "local interactional" properties like this, they can also be used to express "pure interactional" properties (that is, those which contain locally unbound IoC references which appeal to the existence of necessary interactional partners in the environment before they can do their work).
Naturally, all the standard rules of good design continue to apply - one must always use the least powerful capability that can achieve a particular result. But, sometimes, the most appropriate factoring of a design leaves behind one of these "interactional units" - that is, what two elements have in common is a common interaction with a 3rd element.

This can be expressed only by means of our doctrine - in that everything, regardless of what kind of higher-order relation it encodes, is just STATE. "distributeOptions" and "modelRelay", etc. at the end of the day, are just elements in a block of JSON .........